### PR TITLE
Add weather labels and missing constants

### PR DESCRIPTION
**Pull request by SatoMew**
The labels are based on Game Freak's own, which are used in Japanese on the German Ruby debug ROM.

- **なし** _None_
- **はれ** _Clear_
- **はれ2** _Clear 2_
- **あめ** _Rain_
- **ゆき** _Snow_
- **かみなり** _Lightning_
- **きり** _Fog_
- **かざんばい** _Volcanic ash_
- **すなあらし** _Sandstorm_
- **きり2** _Fog 2_
- **かいてい** _Undersea_
- **くもり** _Overcast_
- **はれ3** _Clear 3_
- **おおあめ** _Heavy rain_
- **かいてい2** _Undersea 2_

### CODE REVIEW

**Comment by garakmon** [comment]
I approve this fwiw

**Comment by garakmon** [comment]
though the weather constants do eventually need to end up in include/constants/weather.h so perhaps you can do that now

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This is a disassembly of Pokémon FireRed and LeafGreen.
 
 It builds the following ROM:
 
-* **pokefirered.gba** `sha1: 41cb23d8dccc8ebd7c649cd8fbb58eeace6e2fdc`
+* [**pokefirered.gba**](https://datomatic.no-intro.org/?page=show_record&s=23&n=1616) `sha1: 41cb23d8dccc8ebd7c649cd8fbb58eeace6e2fdc`
```
Comment (PikalaxALT) [inline README.md]: What is this URL for? It's blocked by my firewall

Comment (garakmon) [reply README.md]: it's info about the dump

Comment (SatoMew) [reply README.md]: It's for DAT-O-MATIC's info on the English FireRed v1.0 ROM dump.

**Comment by SatoMew** [comment]
> though the weather constants do eventually need to end up in include/constants/weather.h so perhaps you can do that now that they are documented

OK, will do.

**Pull request merged**